tests/cleo_tests/MemoryOperations/0A8C.txt

```diff
@@ -1,180 +1,88 @@
-else
-    breakpoint "~r~~h~~h~~h~0A8C (write_memory), #3 FAILED!~n~cccccccc dd444444 eeeeeeee Expected~n~%08x %08x %08x Occured" 1@ 2@ 3@
-end
-
-
-wait 0
-// write 4 bytes
-get_var_pointer 2@ {store_to} 0@
-1@ = 0xcccccccc
-2@ = 0xdddddddd
-3@ = 0xeeeeeeee
-
-0A8C: write_memory {address} 0@ {size} 4 {value} 0x11223344 {vp} false // tested opcode
-
-if and
-    1@ == 0xcccccccc
-    2@ == 0x11223344
-    3@ == 0xeeeeeeee
-then
-    trace "~g~~h~~h~0A8C (write_memory), #4 PASSED"
-else
-    breakpoint "~r~~h~~h~~h~0A8C (write_memory), #4 FAILED!~n~cccccccc 11223344 eeeeeeee Expected~n~%08x %08x %08x Occured" 1@ 2@ 3@
-end
-
-
-wait 0
-// write 5 bytes
-get_var_pointer 2@ {store_to} 0@
-1@ = 0xcccccccc
-2@ = 0xdddddddd
-3@ = 0xeeeeeeee
-
-0A8C: write_memory {address} 0@ {size} 5 {value} 0x11223344 {vp} false // tested opcode
-
-if and
-    1@ == 0xcccccccc
-    2@ == 0x44444444
-    3@ == 0xeeeeee44
-then
-    trace "~g~~h~~h~0A8C (write_memory), #5 PASSED"
-else
-    breakpoint "~r~~h~~h~~h~0A8C (write_memory), #5 FAILED! Expected: cccccccc, 44444444, eeeeee44 Actual: %08x %08x %08x" 1@ 2@ 3@
-end
-
-
-wait 0
-// write 7 bytes
-get_var_pointer 2@ {store_to} 0@
-1@ = 0xcccccccc
-2@ = 0xdddddddd
-3@ = 0xeeeeeeee
-
-0A8C: write_memory {address} 0@ {size} 7 {value} 0x11223344 {vp} false // tested opcode
-
-if and
-    1@ == 0xcccccccc
-    2@ == 0x44444444
-    3@ == 0xee444444
-then
-    trace "~g~~h~~h~0A8C (write_memory), #6 PASSED"
-else
-    breakpoint "~r~~h~~h~~h~0A8C (write_memory), #6 FAILED! Expected: cccccccc, 44444444, ee444444 Actual: %08x %08x %08x" 1@ 2@ 3@
-end
+script_name '0A8C'
+test("0A8C (write_memory)", @tests)
+terminate_this_custom_script
 
-wait 0
 
-// write float
-get_var_pointer 2@ {store_to} 0@
-1@ = 0xcccccccc
-2@ = 0xdddddddd
-3@ = 0xeeeeeeee
-4@ = 100.0
+function tests
+    before_each(@before)
+
+    it("should write 0 bytes", @test1)
+    it("should write 1 byte", @test2)
+    it("should write 2 bytes", @test3)
+    it("should write 3 bytes", @test4)
+    it("should write 4 bytes", @test5)
+    it("should write 5 bytes", @test6)
+    it("should write 7 bytes", @test7)
+    it("should write float", @test8)
+
+    return true
+    
+    :before
+        get_var_pointer 2@ {store_to} 0@
+        1@ = 0xcccccccc
+        2@ = 0xdddddddd
+        3@ = 0xeeeeeeee
+    0051: return
+
+    function test1
+        write_memory {address} 0@ {size} 0 {value} 0x11223344 {vp} false
+        assert_eq(1@, 0xcccccccc)
+        assert_eq(2@, 0xdddddddd)
+        assert_eq(3@, 0xeeeeeeee)
+    end
+    
+    function test2
+        write_memory {address} 0@ {size} 1 {value} 0x11223344 {vp} false
+        assert_eq(1@, 0xcccccccc)
+        assert_eq(2@, 0xdddddd44)
+        assert_eq(3@, 0xeeeeeeee)
+    end
+
+    function test3
+        write_memory {address} 0@ {size} 2 {value} 0x11223344 {vp} false
+        assert_eq(1@, 0xcccccccc)
+        assert_eq(2@, 0xdddd3344)
+        assert_eq(3@, 0xeeeeeeee)
```

Comment:
Assert index would be nice too, as in current error message assert source can only be guessed by expected data (which might be identical for few asserts).

Reply:
yes, would be nice to point out the assert location.

Reply:
not sure about explicit index, can be annoying manually tag each assert

Reply:
I mean counted in background. I think you did same thing for test indexing.

Reply:
added index